### PR TITLE
fix(server): [nan-979] Remove dupes if two or more syncs have an error for the connections list page

### PR DESCRIPTION
**Pull request by khaliqgant**
## Describe your changes
If a connection has two or more syncs that have an error it would duplicate that connection on the list page. Use [join lateral](https://www.postgresql.org/docs/9.4/queries-table-expressions.html) > 9.3 postgres version to limit the join.

## Issue ticket number and link
NAN-979

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

File: packages/shared/lib/services/connection.service.ts [hunk 1]
```diff
@@ -528,9 +528,9 @@ class ConnectionService {
     ): Promise<{ id: number; connection_id: string; provider: string; created: string; metadata: Metadata; active_logs: ActiveLogIds }[]> {
         const queryBuilder = db.knex
             .from<Connection>(`_nango_connections`)
-            .leftJoin(ACTIVE_LOG_TABLE, function () {
-                this.on('_nango_connections.id', '=', `${ACTIVE_LOG_TABLE}.connection_id`).andOnVal(`${ACTIVE_LOG_TABLE}.active`, true);
-            })
+            .joinRaw(
+                `LEFT JOIN LATERAL (SELECT * FROM ${ACTIVE_LOG_TABLE} WHERE _nango_connections.id = ${ACTIVE_LOG_TABLE}.connection_id AND ${ACTIVE_LOG_TABLE}.active = true LIMIT 1) ${ACTIVE_LOG_TABLE} ON true`
```
Comment (TBonnin) [inline packages/shared/lib/services/connection.service.ts]: not sure the active logs belong into this query but can you add a comment to explain why we are doing a lateral join subquery.

**Review by TBonnin — APPROVED** [review]

File: packages/shared/lib/services/connection.service.ts (hunk 1 above)
Comment (khaliqgant) [reply packages/shared/lib/services/connection.service.ts]: Added comment with https://github.com/NangoHQ/nango/pull/2263/commits/16291a900206a59a5f0e312338eb711a0ae1d0a6

File: packages/shared/lib/services/connection.service.ts [hunk 2]
```diff
@@ -528,9 +528,14 @@ class ConnectionService {
...
         const queryBuilder = db.knex
             .from<Connection>(`_nango_connections`)
-            .leftJoin(ACTIVE_LOG_TABLE, function () {
-                this.on('_nango_connections.id', '=', `${ACTIVE_LOG_TABLE}.connection_id`).andOnVal(`${ACTIVE_LOG_TABLE}.active`, true);
-            })
+            // A connection can have multiple different syncs that are failing
+            // which means it could have different entries in the active logs table.
+            // We only need to know that the connection has any active logs
+            // and we don't want the connection to show multiple times.
+            // We use a lateral join to get only one active log.
```
Comment (bodinsamuel) [inline packages/shared/lib/services/connection.service.ts]: my take: either a second query or sub select would make all of this clearer

**Review by bodinsamuel — APPROVED** [review]

File: packages/shared/lib/services/connection.service.ts (hunk 2 above)
Comment (khaliqgant) [reply packages/shared/lib/services/connection.service.ts]: Point taken, updated to a sub query

**Pull request merged**